swarm/network/fetcher.go

```diff
@@ -26,27 +26,31 @@ import (
 	"github.com/ethereum/go-ethereum/swarm/storage"
 )
 
-var searchTimeout = 1 * time.Second
+const (
+	defaultSearchTimeout = 1 * time.Second
```

Comment:
Probably best to keep this a `var` so that we can change it from tests.

Reply:
On a second reading, I see that this is exactly what you are trying to fix here. I don't think we want to store the timeout in the `fetcher` - we might have thousands of these. I also see that we have a race by modifying the timeout in tests.

We should probably refactor the tests, rather than store the timeout in every single fetcher.

Reply:
Since we only have two tests, I'd rather we fix the tests, than copy a searchTimeout in every fetcher.